src/load.rs

```diff
@@ -0,0 +1,334 @@
+use std::fs::File;
+use std::io::{BufReader, Read, Seek};
+use std::path::PathBuf;
+use tiff::decoder::{Decoder, DecodingResult};
+use tiff::TiffError;
+
+use snafu::{ResultExt, Snafu};
+
+use crate::color::{ColorError, DicomColorType};
+use crate::metadata::FrameCount;
+use ndarray::{s, Array, Array4};
+
+#[derive(Debug, Snafu)]
+pub enum LoadError {
+    ParseColorType {
+        #[snafu(source(from(TiffError, Box::new)))]
+        source: Box<TiffError>,
+    },
+    #[snafu(display("could not open TIFF file {}", path.display()))]
+    OpenTiff {
+        #[snafu(source(from(std::io::Error, Box::new)))]
+        source: Box<std::io::Error>,
+        path: PathBuf,
+    },
+    #[snafu(display("could not read TIFF file {}", path.display()))]
+    ReadTiff {
+        #[snafu(source(from(TiffError, Box::new)))]
+        source: Box<TiffError>,
+        path: PathBuf,
+    },
+    DecodeImage {
+        #[snafu(source(from(TiffError, Box::new)))]
+        source: Box<TiffError>,
+    },
+    DimensionsError {
+        #[snafu(source(from(TiffError, Box::new)))]
+        source: Box<TiffError>,
+    },
+    UnsupportedColorType {
+        #[snafu(source(from(ColorError, Box::new)))]
+        source: Box<ColorError>,
+    },
+    #[snafu(display("could not convert color type from {:?} to {}", input, target))]
+    ConvertColorType {
+        input: DecodingResult,
+        target: &'static str,
+    },
+    InvalidFrameIndex {
+        frame: usize,
+        num_frames: usize,
+    },
+    SeekToFrame {
+        #[snafu(source(from(TiffError, Box::new)))]
+        source: Box<TiffError>,
+        frame: usize,
+        num_frames: usize,
+    },
+}
+
+/// The order of the channels in the loaded image
+pub enum ChannelOrder {
+    First,
```

Comment:
`First` is a little bit ambiguous because it could mean that `Color` comes first or `Height,Width` comes first. I would rather something like this:
```rust
pub enum ChannelOrder {
    FramesHeightWidthColor,
    FramesColorHeightWidth,
}
```
Or if it needs to be shorter we could use `NHWC` & `NCHW`.

Reply:
I went with the short versions